### PR TITLE
[mmu probing] Add ProbingBase PG counter false env UT coverage

### PR DESCRIPTION
**Pull request by yyynini**
### Description of PR
Add unit test coverage for the real `ProbingBase.setUp()` environment parsing path for `INGRESS_DROP_USE_PG_COUNTER=false`.

This follows reviewer guidance to improve mock UT coverage for existing supported behavior instead of adding edge-case tests that may force unfinished algorithm behavior.

### Type of change
- [ ] Bug fix
- [ ] Testbed and Framework(new/improvement)
- [ ] New Test case
- [x] Test case improvement

#### What is the motivation for this PR?
Improve coverage for the real `ProbingBase.setUp()` path. Existing tests simulated the PG-counter environment parsing logic directly, but did not execute `setUp()` itself.

#### How did you do it?
Added a mock-friendly UT that calls `ProbingBase.setUp()` and verifies `INGRESS_DROP_USE_PG_COUNTER=false` keeps port counter mode, while mocking PTF/switch initialization dependencies.

#### How did you verify/test it?
Mock UT/IT result:

```text
cd tests/saitests/mock/ut
python3 -m pytest -v test_probing_base.py::TestProbingBaseSetUp::test_setUp_applies_ingress_drop_pg_counter_false_env
  -> passed

cd tests/saitests/mock/ut
python3 -m pytest -v .
  -> 446 passed, 1 failed, 4 warnings

cd tests/saitests/mock/it
python3 -m pytest -v .
  -> 75 passed
```

Failure note:

The single full mock UT failure is unrelated to this PR and reproducible on the baseline branch:

```text
test_probing_base.py::TestProbingBaseGetRxPort::test_get_rx_port_wrapper
AssertionError: Should log before and after
assert 1 == 2
```

That baseline test expects two `log_message()` calls from `ProbingBase.get_rx_port()`. Current code emits one `log_message()` call and uses `ProbingObserver.trace()` for the retry/resolution path. This PR only adds `ProbingBase.setUp()` env parsing coverage and does not touch `get_rx_port()`.

### CODE REVIEW

File: tests/saitests/mock/ut/test_probing_base.py [hunk 1]
```diff
@@ -624,6 +624,34 @@ def test_setUp_ingress_drop_pg_counter_default(self):
         assert pb.use_pg_drop_counter is False
         print("[OK] use_pg_drop_counter defaults to False")
 
+    @pytest.mark.order(9373)
```
Comment (XuChen-MSFT) [inline tests/saitests/mock/ut/test_probing_base.py:627]: nit / suggestion: `@pytest.mark.order(9373)` looks inconsistent with this file's 3-digit ascending numbering (910 → 932 → 933-937 [setUp suite] → 938 [TearDown] → 939, 940). Functionally, pytest-order will sort this test after every test with `order ≤ 940`, so it runs at the end of the file rather than next to its setUp() siblings. If this was intentional (e.g., to run last), a one-line comment would help; otherwise it looks like a typo for 938 (with the 938 slot needing a bump). The new test has no shared state so correctness is unaffected either way — just consistency.

File: tests/saitests/mock/ut/test_probing_base.py [hunk 2]
```diff
@@ -624,6 +624,34 @@ def test_setUp_ingress_drop_pg_counter_default(self):
...
+    def test_setUp_applies_ingress_drop_pg_counter_false_env(self):
```
Comment (XuChen-MSFT) [inline tests/saitests/mock/ut/test_probing_base.py:628]: suggestion: the test name and PR rationale narrowly target `INGRESS_DROP_USE_PG_COUNTER=false`, but the assertions also cover unrelated setUp() side-effects (`switch_init` was called, `time.sleep(5)`, `trace` was called, step size unchanged). If `setUp()` is later refactored (e.g., the sleep duration changes, or `switch_init` is wrapped), this test will fail for reasons unrelated to PG-counter handling.

Either keep the test focused (the `assert pb.use_pg_drop_counter is False` alone is enough to prove the env path works end-to-end through the real setUp), or rename to reflect the broader scope (e.g. `test_setUp_real_invocation_smoke_with_pg_counter_false`). I'd lean toward the former for tighter failure attribution.

File: tests/saitests/mock/ut/test_probing_base.py [hunk 3]
```diff
@@ -624,6 +624,34 @@ def test_setUp_ingress_drop_pg_counter_default(self):
...
+        with patch('probing_base.sai_base_test.ThriftInterfaceDataPlane',
+                   MockThriftInterfaceDataPlane):
+            with patch('probing_base.switch_init') as mock_switch_init:
+                with patch('probing_base.time.sleep') as mock_sleep:
+                    with patch('probing_observer.ProbingObserver.trace') as mock_trace:
+                        with patch.dict(os.environ, {'INGRESS_DROP_USE_PG_COUNTER': 'false'}, clear=True):
+                            pb.setUp()
+
+        assert pb.use_pg_drop_counter is False
+        assert pb.POINT_PROBING_STEP_SIZE == 1
```
Comment (XuChen-MSFT) [inline tests/saitests/mock/ut/test_probing_base.py:649]: suggestion: `assert pb.POINT_PROBING_STEP_SIZE == 1` is tautological here — `ConcreteProbingBase.__init__` (top of file) already sets it to 1, and `patch.dict(os.environ, {...}, clear=True)` guarantees the env var is unset, so this assertion would pass even if `setUp()` had no step-size handling at all. Either drop it, or strengthen by setting the env to a non-default in the same `patch.dict`:

```python
with patch.dict(os.environ, {
        'INGRESS_DROP_USE_PG_COUNTER': 'false',
        'POINT_PROBING_STEP_SIZE': '4',
    }, clear=True):
    pb.setUp()
assert pb.POINT_PROBING_STEP_SIZE == 4
```

That would actually exercise the env-driven mutation through the real setUp() path.

**Review by XuChen-MSFT — COMMENTED** [review]
## Code Review Summary

**Verdict**: Approve with minor comments
**Counts**: 0 critical · 0 important · 4 suggestions

### Compliance
| Check | Result |
|---|---|
| DCO sign-off | ✅ |
| PR template | ✅ |
| No SKU exposure | ✅ |
| Test report | ✅ |
| UT/IT changes | ✅ |
| Backport flags | ➖ |
| No force-push | ⚠️ Pre-review rebase only, acceptable |
| CI static analysis | ✅ |

### Highlights
- Suggestion: `@pytest.mark.order(9373)` looks inconsistent with the local 910→940 monotonic numbering — likely a typo (should be 938 or similar). Practical effect: this test runs after every other test in the file rather than next to its setUp siblings.
- Suggestion: `assert pb.POINT_PROBING_STEP_SIZE == 1` is tautological — the constructor already sets it to 1 and the env is cleared, so the assertion would pass even if setUp() had no step-size logic.
- Suggestion: Test name narrowly says "INGRESS_DROP_USE_PG_COUNTER=false" but assertions also cover `switch_init`, `time.sleep`, and `POINT_PROBING_STEP_SIZE`. Consider keeping the test focused (drop the unrelated side-effect assertions) or rename it.
- Suggestion (non-blocking): Coverage gap — the real-setUp() path is only exercised for `=false`. Consider parametrizing the env value to cover `true`/default/etc. through the real path in one shot.

### Findings without specific line anchor
- Coverage gap (parametrize) — see above.

**Comment by bhouse-nexthop** [comment]
Hi @yyynini, we re-triggered CI on this PR and it's still failing. Could you take a look at the latest run and address the failures? Once CI is green we'll proceed with review. Thanks!
